test/source/tests/compose.ts

```diff
@@ -599,13 +599,13 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
       const correctButtonStatusTxt = 'Encrypt, Sign and Send';
       await ComposePageRecipe.showRecipientInput(composePage);
       await composePage.waitAll('@container-cc-bcc-buttons');
-      expect(await composePage.read('@action-send')).to.eq(correctButtonStatusTxt);
+      await composePage.waitForContent('@action-send', correctButtonStatusTxt);
       await composePage.waitAndType(`@input-to`, 'aaaaa\n'); // First enter invalid recipient
-      expect(await composePage.read('@action-send')).to.eq('Re-enter recipient..');
+      await composePage.waitForContent('@action-send', 'Re-enter recipient..');
       await composePage.press('Backspace'); // Delete invalid recipient
-      expect(await composePage.read('@action-send')).to.eq(correctButtonStatusTxt); // check if sent button status is correct
+      await composePage.waitForContent('@action-send', correctButtonStatusTxt); // check if sent button status is correct
       await composePage.waitAndType(`@input-to`, 'mock.only.pubkey@flowcrypt.com\n'); // Now enter correct recipient and check if send button status is correct.
-      expect(await composePage.read('@action-send')).to.eq(correctButtonStatusTxt);
+      await composePage.waitForContent('@action-send', correctButtonStatusTxt);
     }));
```

Comment:
@ioanmo226 reading a value "at a point in time" results in flaky tests, because sometimes the value didn't have a chance to change yet if the test code is faster than the browser extension code in that moment. It could be that 5ms later the right value would have appeared, but the test will fail anyway if written that way.

Usually `composePage.waitForContent` is the more stable way to do it. The test will keep testing the value in a loop repeatedly. If it eventually has the right value, it will succeed, else if it had to wait too long (longer than timeout value) then it will fail.

I've updated it accordingly

Reply:
Got it. Thanks